chain/transaction.go

```diff
@@ -615,3 +616,48 @@ func EstimateUnits(r Rules, actions Actions, authFactory AuthFactory) (fees.Dime
 	}
 	return fees.Dimensions{bandwidth, compute, reads, allocates, writes}, nil
 }
+
+func GenerateTransaction(
+	unitPrices fees.Dimensions,
+	parser Parser,
+	actions Actions,
+	authFactory AuthFactory,
+) (*Transaction, error) {
+	units, err := EstimateUnits(parser.Rules(time.Now().UnixMilli()), actions, authFactory)
+	if err != nil {
+		return nil, err
+	}
+	maxFee, err := fees.MulSum(unitPrices, units)
+	if err != nil {
+		return nil, err
+	}
+	tx, err := GenerateTransactionManual(parser, actions, authFactory, maxFee)
+	if err != nil {
+		return nil, err
+	}
+	return tx, nil
+}
+
+func GenerateTransactionManual(
+	parser Parser,
+	actions Actions,
+	authFactory AuthFactory,
+	maxFee uint64,
+) (*Transaction, error) {
+	// Construct transaction
+	now := time.Now().UnixMilli()
+	rules := parser.Rules(now)
+	base := &Base{
+		Timestamp: utils.UnixRMilli(now, rules.GetValidityWindow()),
+		ChainID:   rules.GetChainID(),
+		MaxFee:    maxFee,
+	}
+
+	// Build transaction
```

Comment:
(unrelated to this PR since it's just copy pasta from the original location) Can we change the comments here to construct and sign and put them in the corresponding places? Alternatively, we could just remove these as this code is simple enough to be self-documenting.